astropy/table/table.py

```diff
@@ -280,12 +282,37 @@ def _rename_column(self, name, new_name):
         if new_name in self:
             raise KeyError(f"Column {new_name} already exists")
 
+        parent_table = self[name].info.parent_table
+        if (parent_table is not None
+                and parent_table.hidden_columns()
+                and name in parent_table.hidden_columns()):
+            hc = list(parent_table.hidden_columns())
+            hc[hc.index(name)] = new_name
+            parent_table.hidden_columns = hc
+
         mapper = {name: new_name}
         new_names = [mapper.get(name, name) for name in self]
         cols = list(self.values())
         self.clear()
         self.update(list(zip(new_names, cols)))
 
+    def _remove_hidden_column(self, name):
+        parent_table = self[name].info.parent_table
+        if parent_table is not None and parent_table.hidden_columns():
+            parent_table.hidden_columns.remove(name)
```

Comment:
Don't you need to check that the `name` was hidden before? I think `.remove` would give an error.

Reply:
Yup, that is fixed in the latest version.